pkg/directory/gh.go

```diff
@@ -0,0 +1,363 @@
+package directory
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/google/go-github/github"
+	"github.com/pkg/errors"
+	"golang.org/x/oauth2"
+)
+
+const (
+	// GHAllTeam containing all users in GitHub
+	GHAllTeam = "all"
+)
+
+// UsersService holds methods used in the GitHub UsersService for easier testing
+type UsersService interface {
+	Get(context.Context, string) (*github.User, *github.Response, error)
+}
+
+// GH hosts a client for accessing GH as well as cached Member and Team lists
+type GH struct {
+	*github.Client
+
+	UsersService UsersService
+	Info
+}
+
+// NewGitHub returns an initialized GitHub client to the caller and stored GH members and teams
+func NewGitHub(org string) (*GH, error) {
+	ctx := context.Background()
+	client := &GH{}
+
+	token, ok := os.LookupEnv("GITHUB_TOKEN")
+	if !ok {
+		return client, errors.New("GITHUB_TOKEN not set")
+	}
+	ts := oauth2.StaticTokenSource(
+		&oauth2.Token{AccessToken: token},
+	)
+	tc := oauth2.NewClient(ctx, ts)
+	client.Client = github.NewClient(tc)
+	client.UsersService = client.Client.Users
+	client.Org = org
+
+	if err := client.getMembersAndTeams(); err != nil {
+		return client, err
+	}
+	return client, nil
+}
+
+func (g *GH) getMembersAndTeams() error {
+	update := false
+
+	_, err := os.Stat(cacheDir)
+	if err != nil {
+		os.MkdirAll(cacheDir, os.ModePerm)
+	}
+
+	membersFile := path.Join(cacheDir, "members")
+	mfInfo, err := os.Stat(membersFile)
+	if err != nil || time.Since(mfInfo.ModTime()).Minutes() > cacheTTL {
+		update = true
```

Comment:
```go
update = time.Since(mfInfo.ModTime()).Minutes() > cacheTTL
```

Reply:
I'm not sure why that would be better. Could you explain?

Reply:
Nvm. Disregard.